virtool/references/db.py

```diff
@@ -239,64 +240,82 @@ def __init__(self, app, task_id):
         super().__init__(app, task_id)
 
         self.steps = [
+            self.remove_directory,
             self.remove_indexes,
             self.remove_unreferenced_otus,
             self.remove_referenced_otus
         ]
 
-    async def remove_indexes(self):
-        ref_id = self.context["ref_id"]
+        self.non_existent_references = []
+
+    async def remove_directory(self):
+        path = os.path.join(
+            self.app["settings"]["data_path"],
+            "references"
+        )
 
-        await self.db.indexes.delete_many({
-            "reference.id": ref_id
+        reference_ids = os.listdir(path)
+        existent_references = await self.db.references.distinct("_id", {
+            "_id": {
+                "$in": reference_ids
+            }
         })
+        self.non_existent_references = [ref_id for ref_id in reference_ids if ref_id not in existent_references]
 
-    async def remove_unreferenced_otus(self):
-        ref_id = self.context["ref_id"]
+        for dir_name in self.non_existent_references:
+            shutil.rmtree(os.path.join(path, dir_name))
```

Comment:
Needs to run in `ThreadPoolExecutor`. If file system access is slow, this will block the whole application until `rmtree()` returns.